packages/beacon-node/test/unit/chain/rewards/blockRewards.test.ts

```diff
@@ -89,7 +89,10 @@ describe("chain / rewards / blockRewards", () => {
       // Populate tree root caches of the state
       state.hashTreeRoot();
       cachedStateAltairPopulateCaches(state);
-      const calculatedBlockReward = await computeBlockRewards(block.message, state as CachedBeaconStateAllForks);
+      const calculatedBlockReward = await computeBlockRewards(
+        block.message,
+        state as unknown as CachedBeaconStateAllForks
```

Comment:
![image](https://github.com/ChainSafe/lodestar/assets/38436224/3b5b61cd-e2ff-45e0-86a7-757f7ec22f29)

Reply:
But jokes aside, what happened on electra branch that's states between forks can't even be type casted anymore? Doing `as unknown as someOtherType` is almost the same as just doing `as any` which means we have disabled typescript at this point.


On unstable branch this is not an issue
https://github.com/ChainSafe/lodestar/blob/7fadd303fad8679ee2b7462747b9d9fc33dafb23/packages/beacon-node/test/unit/chain/rewards/blockRewards.test.ts#L140

so not sure if this is the right fix here

Reply:
agree, should add a comment to look into that later. Otherwise I'm afraid the same pattern will be applied everywhere

Reply:
> > In electra-fork, CachedBeaonStateElectra is introduced to CachedBeaconStateAllForks which adds more difference, this is why check-types complains about "insufficiently overlaps". See the job [log](https://github.com/ChainSafe/lodestar/actions/runs/8554667832/job/23473887770?pr=6528)
> 
> Not sure if this explains it, Deneb, or even Bellatrix state have sufficient difference, adding more properties shouldn't break the type cast. I would assume it breaks if you modify or remove a property

I just spent some time looking into this. It looks like it's because in electra-fork branch there is a newly added field in `EpochCache` named `private historialValidatorLengths`. 

So when casting from `CachedBeaconStateAltair` to `CachedBeaconStateAllForks`, instead of comparing the two structurally, it's now comparing nominally because of the private property. 

I just removed the private modifier of `historialValidatorLengths` in favour of cleaner code. 

Sorry for the incorrect explanation I previous made

Reply:
> So when casting from CachedBeaconStateAltair to CachedBeaconStateAllForks, instead of comparing the two structurally, it's now comparing nominally because of the private property.

Thanks for investigating that, this is much better than having to cast `as unknown`. Why marking the property as `private` causes issues is a bit strange though, it should only check the pubic interface.

Might be related to a typescript bug https://github.com/microsoft/TypeScript/issues/37621, @matthewkeil this is similar to what you observed on the shuffling refactor branch?